modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/util/v1_0/VirtualSettingsUtil.java

```diff
@@ -0,0 +1,362 @@
+
+			if (fileEntry != null) {
+				return true;
+			}
+
+			return false;
+		}
+		catch (PortalException portalException) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(portalException);
+			}
+
+			return false;
+		}
+	}
+
+	private static int _getActivationStatus(int activationStatus) {
+		if (ArrayUtil.contains(
+				VirtualCPTypeConstants.ACTIVATION_STATUSES, activationStatus)) {
+
+			return activationStatus;
+		}
+
+		return CommerceOrderConstants.ORDER_STATUS_COMPLETED;
+	}
+
+	private static CPDefinitionVirtualSetting _updateVirtualSettings(
+			CPDefinitionVirtualSetting cpDefinitionVirtualSetting,
+			ProductVirtualSettings productVirtualSettings,
+			CPDefinitionVirtualSettingService cpDefinitionVirtualSettingService,
+			UniqueFileNameProvider uniqueFileNameProvider,
+			ServiceContext serviceContext)
+		throws Exception {
+
+		long attachmentFileEntryId = 0;
+		String attachmentUrl = _validateUrl(productVirtualSettings.getUrl());
+
+		if (Validator.isNull(attachmentUrl)) {
+			if (Validator.isNull(productVirtualSettings.getAttachment())) {
+				attachmentUrl = cpDefinitionVirtualSetting.getUrl();
+			}
+			else {
+				attachmentFileEntryId = _addFileEntry(
+					productVirtualSettings.getAttachment(), attachmentUrl,
+					uniqueFileNameProvider, serviceContext);
+			}
+
+			if (attachmentFileEntryId == 0) {
+				attachmentFileEntryId =
+					cpDefinitionVirtualSetting.getFileEntryId();
+			}
+		}
+
+		Long duration = productVirtualSettings.getDuration();
+
+		if (Validator.isNotNull(duration)) {
+			duration = TimeUnit.DAYS.toMillis(duration);
+		}
+
+		boolean useSample = GetterUtil.get(
+			productVirtualSettings.getUseSample(),
+			cpDefinitionVirtualSetting.isUseSample());
+		String sampleAttachmentUrl = null;
+		long sampleFileEntryId = 0;
+
+		if (useSample) {
+			sampleAttachmentUrl = _validateUrl(
+				productVirtualSettings.getSampleUrl());
+
+			if (Validator.isNull(sampleAttachmentUrl)) {
+				if (Validator.isNull(
+						productVirtualSettings.getSampleAttachment())) {
+
+					sampleAttachmentUrl =
+						cpDefinitionVirtualSetting.getSampleUrl();
+				}
+				else {
+					sampleFileEntryId = _addFileEntry(
+						productVirtualSettings.getSampleAttachment(),
+						sampleAttachmentUrl, uniqueFileNameProvider,
+						serviceContext);
+				}
+
+				if (sampleFileEntryId == 0) {
+					sampleFileEntryId =
+						cpDefinitionVirtualSetting.getSampleFileEntryId();
+				}
+			}
+		}
+
+		Map<Locale, String> termsOfUseContentMap = null;
+		long termsOfUseJournalArticleId = 0;
+		boolean termsOfUseRequired = GetterUtil.get(
+			productVirtualSettings.getTermsOfUseRequired(),
+			cpDefinitionVirtualSetting.isTermsOfUseRequired());
+
+		if (termsOfUseRequired) {
+			termsOfUseContentMap = LanguageUtils.getLocalizedMap(
+				productVirtualSettings.getTermsOfUseContent());
+			termsOfUseJournalArticleId = GetterUtil.get(
+				productVirtualSettings.getTermsOfUseJournalArticleId(), 0);
```

Comment:
You don't need to specify a default value of 0

Reply:
Done